Practice Areas/Engineering/RFCs/2019-09-02 Vets-Website CMS Integration.md

```diff
@@ -0,0 +1,85 @@
+
+We could also define some build-breaking criteria, where we have reached a certain threshold that our is determined by our build to not be deployable. For example, perhaps 100 broken links is _okay_ (considering the website will eventually be composed of many, many pages), but if there are 100 broken links to the same resource (such as `/health-care`), then this is not an acceptable state for the website to be deployed, because a key resource is missing. In this case, the build or deployment will fail.
+
+If there are other evaluations that we wish to perform at this step, we could group that with the broken-link checker into a general "health check" stage. As a later iteration, we could explore emailing the results of the health check to an email listing for CMS users, so that the alert is passed to the people managing the content.
+
+### Accessibility tests
+
+__The accessibility sub-step of our Integration step in our CI should be removed and replaced by a warning banner from our CMS Preview Server__. Currently, the accessibility step works by stepping throughout the generated website sitemap and executing an accessibility check via `axe-core` on each page. If one page fails, the whole build fails. Because HTML containing an accessibility issue can be entered into a CMS WYSIWYG, it has the same issues as the broken-link checker.
+
+Currently, we already have accessibility tests executing in our application's E2E (end-to-end) tests via the `axe-core` module. This means that the React portion of the website should already largely meet accessibility standards, or at least in what can be automated. This means that only content pages need to be evaluated.
+
+Instead of failing a build when an accessibility issue has been saved into the CMS, we should introduce "warnings" into the CMS Preview Server - the CMS Preview Server injects and executes `axe-core`, then when an issue(s) is found, an alert banner is rendered to the author with the results. We could also enable [best practices](https://www.deque.com/axe/axe-for-web/documentation/api-documentation/) to help encourage higher quality accessibility practices than we do currently. 
+
+This proposal has already been captured into [issue 1794](https://github.com/department-of-veterans-affairs/va.gov-team/issues/1794).
+
+
+## Risks
+_What are the risks of the proposed changes?_
+
+The largest risk by far is degraded website quality overtime. If broken links become non-breaking and the parties responsible for maintaining the content aren't made aware, we could see broken links and small accessibility issues becoming widespread throughout the site.
```

Comment:
Yeah, I feel like adding rules and thresholds to allow a certain level of "badness" into the site could put VA.gov in a position where the same broken links are never fixed, which essentially puts us back in the same situation that we're in now, but where the status quo is > 0 broken links and accessibility issues.

(I can probably expand on that if that doesn't make sense right now.)